lib/i18n/backend/base.rb

```diff
@@ -29,7 +29,30 @@ def translate(locale, key, options = EMPTY_HASH)
         raise InvalidLocale.new(locale) unless locale
         return nil if key.nil? && !options.key?(:default)
 
-        entry = lookup(locale, key, options[:scope], options) unless key.nil?
+        entry = nil
+
+        unless key.nil?
+          normalized_key = I18n.normalize_keys(locale, key, options[:scope])[1..-1].join(I18n.config.default_separator).to_sym
+          deprecated_key = I18n.config.deprecations.key(normalized_key)
+
+          # Does this key supersede a deprecated one?
+          if deprecated_key
+            # Try to use up the old key, if present.
+            entry = lookup(locale, deprecated_key, nil, options)
+          end
+
+          entry ||= lookup(locale, key, options[:scope], options)
+
+          if entry.nil?
+            # Is this key deprecated by another?
+            new_key = I18n.config.deprecations[normalized_key]
+
+            if new_key
+              puts "DEPRECATION WARNING: #{normalized_key} is deprecated, use #{new_key} instead"
```

Comment:
Consider using `warn` (printing to stderr) instead of puts, as it is more common for deprecations.
You might also want to use `warn(…, uplevel: 1)` for rubies that support it, but you need to weight the checks on ruby version vs. the usability improvement of knowing when the error is originated.